src/lem/circuit.rs

```diff
@@ -56,9 +58,157 @@ use super::{
     Block, Ctrl, Func, Op, Tag, Var,
 };
 
+#[derive(Clone)]
+pub struct MultiFrame<'a, F: LurkField> {
+    pub func: &'a Func,
+    pub store: Option<&'a Store<F>>,
+    pub input: Option<Vec<Ptr<F>>>,
+    pub output: Option<Vec<Ptr<F>>>,
+    pub frames: Option<Vec<Frame<F>>>,
+    pub cached_witness: Option<WitnessCS<F>>,
+    pub count: usize,
+}
+
+impl<'a, F: LurkField> MultiFrame<'a, F> {
+    pub fn blank(func: &'a Func, count: usize) -> Self {
+        Self {
+            func,
+            store: None,
+            input: None,
+            output: None,
+            frames: None,
+            cached_witness: None,
+            count,
+        }
+    }
+
+    pub fn get_store(&self) -> &Store<F> {
+        self.store.expect("store missing")
+    }
+
+    pub fn from_frames(
+        func: &'a Func,
+        count: usize,
+        frames: &[Frame<F>],
+        store: &'a Store<F>,
+    ) -> Vec<Self> {
+        // `count` is the number of `Frames` to include per `MultiFrame`.
```

Comment:
We might consider calling this `reduction_count` — since I think we've standardized that as an unambiguous name. Unfortunately, `count` is indeed a very overloaded name, hence the need for this comment.